### PR TITLE
Add tr() descriptors for Taproot output scripts

### PR DESCRIPTION
**Pull request by achow101**
Implements `tr()` descriptors

### CODE REVIEW

File: hwilib/descriptor.py [hunk 1]
```diff
@@ -209,21 +212,21 @@ def __lt__(self, other: 'PubkeyProvider') -> bool:
 class Descriptor(object):
     r"""
     An abstract class for Descriptors themselves.
-    Descriptors can contain multiple :class:`PubkeyProvider`\ s and no more than one ``Descriptor`` as a subdescriptor.
+    Descriptors can contain multiple :class:`PubkeyProvider`\ s and multipe ``Descriptor`` as subdescriptors.
```
Comment (prusnak) [inline hwilib/descriptor.py:215]:
```suggestion
    Descriptors can contain multiple :class:`PubkeyProvider`\ s and multiple ``Descriptor`` as subdescriptors.
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: hwilib/descriptor.py (hunk 1 above)
Comment (achow101) [reply hwilib/descriptor.py]: Done

**Review by Sjors — COMMENTED** [review]
6dedec4c2f8374fb93501152ae3aa327ec61084b looks alright, but I did not study the merkle tree parsing and writing code much.

You could add some more test vectors (from Bitcoin Core?), which are needed for https://github.com/bitcoin/bips/blob/master/bip-0386.mediawiki anyway.

File: hwilib/descriptor.py [hunk 2]
```diff
@@ -430,12 +430,14 @@ def _parse_descriptor(desc: str, ctx: '_ParseDescriptorContext') -> 'Descriptor'
     :raises: ValueError: if the descriptor is malformed
     """
     func, expr = _get_func_expr(desc)
-    if func == "pkh":
```
[the remark this answers is not in the document]
Comment (achow101) [reply hwilib/descriptor.py:433]: Done

**Comment by Sjors** [comment]
Thanks for the changes in 6539d6c4967f74fa2a2bcc022067d3c9dd3cb9c2. I still think more test vectors would be useful here.

**Comment by achow101** [comment]
Core doesn't really have any test vectors that we can use.

**Comment by prusnak** [comment]
> Core doesn't really have any test vectors that we can use.

Feel free to use the test vectors from:
* https://github.com/satoshilabs/slips/blob/master/slip-0014.md#bitcoin-taproot--p2tr--bip86
* https://github.com/satoshilabs/slips/blob/master/slip-0014.md#bitcoin-testnet-taproot--p2tr--bip86

**Comment by Sjors** [comment]
utACK d2d360c4cc7db84afea41c56497cf33f0726b136

**Pull request merged**